File: code_samples/typescript/error_handling_error_propagation.ts
/**
 * Error Handling: Error Propagation
 * AI/ML Training Sample
 */

interface IErrorPropagation {
    data: string;
    process(input: string): void;
    validate(): boolean;
}

class ErrorPropagation implements IErrorPropagation {
    data: string;
    
    constructor() {
        this.data = "";
    }
    
    process(input: string): void {
        this.data = input;
    }
    
    getData(): string {
        return this.data;
    }
    
    validate(): boolean {
        return this.data.length > 0;
    }
}

// Example usage
const instance = new ErrorPropagation();
instance.process("example");
console.log(`Data: ${instance.getData()}`);
console.log(`Valid: ${instance.validate()}`);

export { ErrorPropagation, IErrorPropagation };

File: code_samples/r/error_handling_error_propagation.r
# Error Handling: Error Propagation
# AI/ML Training Sample

ErrorPropagation <- setRefClass(
    "ErrorPropagation",
    fields = list(data = "character"),
    methods = list(
        initialize = function() {
            data <<- ""
        },
        process = function(input) {
            data <<- input
        },
        getData = function() {
            return(data)
        },
        validate = function() {
            return(nchar(data) > 0)
        }
    )
)

# Example usage
instance <- ErrorPropagation$new()
instance$process("example")
cat("Data:", instance$getData(), "\n")
cat("Valid:", instance$validate(), "\n")
